traitlets/traitlets.py

```diff
@@ -359,13 +358,21 @@ def __init__(self, default_value=Undefined, allow_none=None, read_only=None, **m
             )
 
         if len(metadata) > 0:
+            warn("metadata should be set using the .tag() method, e.g., Int().tag(key1='value1', key2='value2')", DeprecationWarning, stacklevel=2)
```

Comment:
This should include the metadata keys being set. Most people who encounter this message won't have any idea what values are metadata and what aren't.